core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -189,16 +199,68 @@ object ConsumerGroupCommand extends Logging {
     }
 
     def listGroups(): Unit = {
-      if (opts.options.has(opts.stateOpt)) {
-        val stateValue = opts.options.valueOf(opts.stateOpt)
-        val states = if (stateValue == null || stateValue.isEmpty)
-          Set[ConsumerGroupState]()
-        else
-          consumerGroupStatesFromString(stateValue)
-        val listings = listConsumerGroupsWithState(states)
-        printGroupStates(listings.map(e => (e.groupId, e.state.get.toString)))
-      } else
+      val includeType = opts.options.has(opts.typeOpt)
+      val includeState = opts.options.has(opts.stateOpt)
+
+      val groupInfoMap = mutable.Map[String, (String, String)]()
+
+      if (includeType || includeState) {
+        val types = getTypeValues()
+        val states = getStateValues()
+        val listings = {
+          listConsumerGroupsWithFilters(types, states)
+        }
+
+        listings.foreach { listing =>
+          val groupId = listing.groupId
+          val groupType = listing.groupType().orElse(GroupType.UNKNOWN).toString
+          val state = listing.state().orElse(ConsumerGroupState.UNKNOWN).toString
+          groupInfoMap.update(groupId, (groupType, state))
```

Comment:
I wonder if converting the listing to a map is really necessary. I understand that it was there before. However, it seems that we could just pass the listing to `printGroupInfo` and avoid it. What do you think?